lib/rouge/formatters/html_legacy.rb

```diff
@@ -28,8 +28,10 @@ def initialize(opts={})
         @formatter = opts[:inline_theme] ? HTMLInline.new(opts[:inline_theme])
                    : HTML.new
 
-
-        @formatter = HTMLTable.new(@formatter, opts) if opts[:line_numbers]
+        if opts[:line_numbers]
+          table_formatter_class = opts[:line_table] ? HTMLLineTable : HTMLTable
+          @formatter = table_formatter_class.new(@formatter, opts)
```

Comment:
This is perhaps another dumb question but why do you need to change the internals of the legacy formatter? Can't it just do its own thing and then people that want to use your new formatter use it?

Reply:
:slightly_smiling_face: For future reference, I do not consider any question to be a dumb question.

I have to change the legacy formatter because it was designed to be just backwards-compatible.  New formatters have to be internally linked.

Reply:
Oh, I think there are sometimes some pretty dumb questions :P The main thrust of my opposition to this change is expressed further below.